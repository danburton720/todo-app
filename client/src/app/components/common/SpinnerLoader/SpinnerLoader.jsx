import React from 'react';
import Loader from 'react-loader-spinner';

import styles from './spinnerLoader.scss';

export default class SpinnerLoader extends React.Component {
    render() {
        return (
            <div className={styles.spinnerLoaderWrapper}>
                <Loader
                    type='Oval'
                    color='#0066ff'
                    height={50}
                    width={50}
                    timeout={100000}
                />
            </div>
        )
    }
}