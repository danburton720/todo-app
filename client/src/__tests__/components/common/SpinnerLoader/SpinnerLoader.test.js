import React from 'react';
import {shallow} from 'enzyme';
import SpinnerLoader from '../../../../app/components/common/SpinnerLoader/SpinnerLoader';

describe('SpinnerLoader', () => {
    let shallowWrapper;
    let props;
    const wrapper = () => {
        if (!shallowWrapper) {
            shallowWrapper = shallow(<SpinnerLoader {...props}/>);
        }
        return shallowWrapper;
    };

    beforeEach(() => {
        props = {};
        shallowWrapper = undefined;
    });

    describe('render', () => {
        it('should return a div with a Loader component', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
});