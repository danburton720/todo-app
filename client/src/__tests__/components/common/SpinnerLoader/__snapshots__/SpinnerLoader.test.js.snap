// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinnerLoader render should return a div with a Loader component 1`] = `
<div
  className="spinnerLoaderWrapper"
>
  <Loader
    className=""
    color="#0066ff"
    height={50}
    style={Object {}}
    timeout={100000}
    type="Oval"
    visible={true}
    width={50}
  />
</div>
`;
